File: backend/app/schemas/submission.py
"""Chore submission request/response schemas."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class SubmissionCreateRequest(BaseModel):
    """Body for POST /submissions. `member_id` comes from the X-Active-Member header."""

    chore_id: uuid.UUID
    notes: str | None = Field(default=None, max_length=2000)


class SubmissionRead(BaseModel):
    """Public view of a submission."""

    model_config = ConfigDict(from_attributes=True)

    submission_id: uuid.UUID
    chore_id: uuid.UUID
    member_id: uuid.UUID
    device_id: uuid.UUID
    household_id: uuid.UUID
    completed_at: datetime
    approved: bool
    rejection_reason: str | None
    notes: str | None


class SubmissionRejectRequest(BaseModel):
    """Body for PATCH /submissions/{id}/reject."""

    reason: str = Field(min_length=1, max_length=2000)

File: flutter_app/lib/core/models/submission.dart
class Submission {
  Submission({
    required this.submissionId,
    required this.choreId,
    required this.memberId,
    required this.deviceId,
    required this.householdId,
    required this.completedAt,
    required this.approved,
    this.rejectionReason,
    this.notes,
  });

  final String submissionId;
  final String choreId;
  final String memberId;
  final String deviceId;
  final String householdId;
  final DateTime completedAt;
  final bool approved;
  final String? rejectionReason;
  final String? notes;

  factory Submission.fromJson(Map<String, dynamic> json) => Submission(
        submissionId: json['submission_id'] as String,
        choreId: json['chore_id'] as String,
        memberId: json['member_id'] as String,
        deviceId: json['device_id'] as String,
        householdId: json['household_id'] as String,
        completedAt: DateTime.parse(json['completed_at'] as String),
        approved: json['approved'] as bool,
        rejectionReason: json['rejection_reason'] as String?,
        notes: json['notes'] as String?,
      );
}
